src/cracker.c

```diff
@@ -63,9 +64,14 @@ void crack()
 		generate_pins();
 
 		/* Restore any previously saved session */
-		if(get_static_p1() == NULL)
+		if(get_static_p1() == NULL || !get_pin_string_mode())
 		{
-			restore_session();
+			session_restored = restore_session();
```

Comment:
... if we move the declaration to here, it would be clearer that it's only used in this scope...

e.g. `int session_restored = ...`

or even more concise:
`if (restore_session()) { return }`

Reply:
I guess these is better
`if (restore_session() == -1) return;`

Do I create new commit or amend your last commit?

Reply:
please create a new one, thanks